src/Atmos/Model/moisture.jl

```diff
@@ -239,3 +240,128 @@ function flux_diffusive!(moist::EquilMoist, flux::Grad, state::Vars, d_q_tot)
     flux.ρu += d_q_tot .* state.ρu'
     flux.moisture.ρq_tot += d_q_tot * state.ρ
 end
+"""
+  NonEquilMoist
+
+  Assumes that there isn't any thermodynamic equilibrium
+  
+  Adds addtional moisture variable for cloud water ρq_liq and cloud ice ρq_ice::FT that are transported in the same way as ρq_tot
+
+  The equilibrium phase partition is calculated and used as a reference to compute the source term for the cloud phase.
+
+"""
+
+struct NonEquilMoist <: MoistureModel end
+
+vars_state(::NonEquilMoist, FT) = @vars(ρq_tot::FT, ρq_liq::FT, ρq_ice::FT)
+vars_gradient(::NonEquilMoist, FT) =
+    @vars(q_tot::FT, q_liq::FT, q_ice::FT, h_tot::FT)
+vars_diffusive(::NonEquilMoist, FT) = @vars(
+    ∇q_tot::SVector{3, FT},
+    ∇q_liq::SVector{3, FT},
+    ∇q_ice::SVector{3, FT}
+)
+vars_aux(::NonEquilMoist, FT) =
+    @vars(temperature::FT, θ_v::FT, q_liq::FT, q_ice::FT, p::FT, src_qliq::FT)
+
+@inline function atmos_nodal_update_aux!(
+    moist::NonEquilMoist,
+    atmos::AtmosModel,
+    state::Vars,
+    aux::Vars,
+    t::Real,
+)
+    TS = thermo_state(moist, atmos.orientation, state, aux)
+    aux.moisture.temperature = air_temperature(TS)
+    q_p_sat = PhasePartition_equil(
+        air_temperature(TS),
+        state.ρ,
+        state.moisture.ρq_tot,
+    )
+    aux.moisture.src_qliq = conv_q_vap_to_q_liq(q_p_sat, PhasePartition(TS))
+    aux.moisture.θ_v = virtual_pottemp(TS)
+    aux.moisture.q_liq = PhasePartition(TS).liq
```

Comment:
I'm not sure about here though. Your new value of `q_liq` and `q_ice` should be coming from the `conv_q_vap_to_q_liq` source function and not from `PhasePartition`. Since its a `NonEquil` setup we can't diagnose it anymore. We have to calculate it based on state and sources

Reply:
You're right. But here aux.moisture.q_liq is not a state variable, it's likely that line 283 isn't really necessary but is nice when processing outputs. The update to q_liq is done through source.jl at : https://github.com/climate-machine/CLIMA/blob/18b64dc37507cbbb69577f8ac074a18ef01983e2/src/Atmos/Model/source.jl#L173

Reply:
essentially 281 calculates the source for the next time step.
283 sets aux.moisture.q_liq from the current time step which is nice to have as an output since the state variable is state.moisture.ρq_liq ( we can have these both in our output files and not have to divide by ρ in post) Though I understand how this might be confusing.